### DISCUSSION TITLE
Proposal: add setQuery function with staleTime to queryCache

### DISCUSSION THREAD

**Discussion by loiacon** [TanStack/query#451] [General]
I think, today, it's not possible to create a query in `queryCache` with `staleTime`:
https://codesandbox.io/s/tender-kare-45cke

**Proposal**
Create `setQuery` function inside `queryCache` that accepts `staleTime`:
```ts
const getTodos = async () => {
  const cachedData = queryCache.getQueryData("todos")
  if (cachedData) {
    return cachedData
  }

  const { data, error } = await queryCache.setQuery("todos", fetchTodos, {
    staleTime: 3000
  })
  if (error) {
    ...
  }
  return data
}
```

**Comment by tannerlinsley** [comment]
This is a good idea. I would accept a PR + tests to enable this functionality :)
